paddle/framework/operator.h

```diff
@@ -406,7 +474,7 @@ class OperatorWithKernel : public OperatorBase {
   }
 
  protected:
-  virtual void InferShape(const InferShapeContext& ctx) const = 0;
+  virtual void InferShape(InferShapeContextBase* ctx) const = 0;
```

Comment:
This file should be changed a lot. `void OperatorBase::InferShape(const Scope& scope)` should be removed. `void InferShape(InferShapeContextBase* ctx) const` should be public.

Reply:
Yes, This will be done in next pr that will modify all the python related code.